src/librbd/io/CopyupRequest.cc

```diff
@@ -401,7 +402,7 @@ void CopyupRequest<I>::copyup() {
 
   ldout(cct, 20) << dendl;
 
-  bool copy_on_read = m_pending_requests.empty();
+  bool copy_on_read = m_pending_requests.empty() && !m_deep_copied;
```

Comment:
@dillaman Just wondering, couldn't `is_deep_copy()` be used instead of introducing a new variable?

Reply:
I wanted to avoid any possibility of a race since if the `migration_info` header changes but we already performed a deep-copy, the `copyup` step will incorrectly overwrite the HEAD revision.